File: 8.Best-Practice-e-trucchi-avanzati/objc/CM_Picker/CM_Picker WatchKit Extension/SequenceController.m
//
//  SequenceController.m
//  CM_Picker
//
//  Created by WhiteTiger "sgama la rete" on 28/02/16.
//  Copyright © 2016 WhiteTiger "sgama la rete". All rights reserved.
//

#import "SequenceController.h"

@interface SequenceController ()

@property (nonatomic, weak) IBOutlet WKInterfacePicker *picker;

@end

@implementation SequenceController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
    
    [self createPickerItems];
}

- (void)willActivate {
    // This method is called when watch view controller is about to be visible to user
    [super willActivate];
}

- (void)didDeactivate {
    // This method is called when watch view controller is no longer visible
    [super didDeactivate];
}

- (void)createPickerItems {
    NSMutableArray<WKPickerItem *> *pickerItems = [NSMutableArray array];
    
    for (int i=0; i<29; i++) {
        WKPickerItem *pickerItem = [WKPickerItem new];
        
        [pickerItem setContentImage:[WKImage imageWithImageName:[NSString stringWithFormat:@"frame_%d", i]]];
        [pickerItems addObject:pickerItem];
    }
    
    [self.picker setItems:pickerItems];
}

@end

File: 8.Best-Practice-e-trucchi-avanzati/swift/CM_Picker/CM_Picker WatchKit Extension/SequenceController.swift
//
//  Copyright © 2015 WhiteTiger "sgama la rete". All rights reserved.
//

import WatchKit
import Foundation

class SequenceController: WKInterfaceController {

    @IBOutlet var picker: WKInterfacePicker!
    
    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
        
        self.createPickerItems()
    }
    
    override func willActivate() {
        super.willActivate()
    }
    
    override func didDeactivate() {
        super.didDeactivate()
    }
    
    // MARK: Private Methods
    
    private func createPickerItems() {
        var pickerItems = [WKPickerItem]()
        
        for index in 0...29 {
            let pickerItem = WKPickerItem()
            
            pickerItem.contentImage = WKImage(imageName: "frame_\(index).png")
            pickerItems.append(pickerItem)
        }
        
        picker.setItems(pickerItems)
    }
    
    @IBAction func actionPicker(value: Int) {
        
    }

}
